src/helpers/errors.js

```diff
@@ -13,37 +13,44 @@ export const createLibraryNotSupportedError = (error: Error) =>
     `Currently the only supported library to search by text is "react-native".\n\n${error.message}`
   );
 
+export const prepareErrorMessage = (error: Error) =>
+  // Strip info about custom predicate
+  error.message.replace(/ matching custom predicate[^]*/gm, '');
+
+export const createQueryByError = (error: Error, callsite: Function) => {
+  if (error.message.includes('No instances found')) {
+    return null;
+  }
+  throw new ErrorWithStack(error.message, callsite);
+};
+
 const warned = {
   getByName: false,
   getAllByName: false,
   queryByName: false,
   queryAllByName: false,
+
+  getByProps: false,
+  getAllByProps: false,
+  queryByProps: false,
+  queryAllByProps: false,
+
+  getByType: false,
+  getAllByType: false,
+  queryByType: false,
+  queryAllByType: false,
 };
 
-export const logDeprecationWarning = (
-  deprecatedFnName: string,
-  alternativeFnName: string
-) => {
-  if (warned[deprecatedFnName]) {
+export function printDeprecationWarning(functionName: string) {
```

Comment:
let's rename this to `printMigrationToUnsafeWarning` and `logDeprecationWarning` -> `printDeprecationWarning`, essentially keeping them both